zds/featured/views.py

```diff
@@ -209,6 +236,73 @@ def post(self, request, *args, **kwargs):
         return redirect(reverse('featured-resource-list'))
 
 
+class FeaturedRequestedList(ZdSPagingListView):
+    """
+    Displays the list of featured resources.
+    """
+
+    context_object_name = 'featured_request_list'
+    paginate_by = settings.ZDS_APP['featured_resource']['request_per_page']
+    template_name = 'featured/list_requests.html'
+
+    def get_queryset(self):
+        queryset = FeaturedRequested.objects\
+            .prefetch_related('content_object')\
+            .annotate(num_vote=Count('users_voted'))\
+            .order_by('-num_vote')
+
+        type_featured_request = self.request.GET.get('type', None)
+        if type_featured_request == 'topic':
+            queryset = queryset.filter(type='TOPIC')
+        elif type_featured_request == 'content':
+            queryset = queryset.filter(type='CONTENT')
+
+        if type_featured_request == 'ignored':
+            queryset = queryset.filter(rejected=True)
+        else:
+            queryset = queryset.filter(rejected=False)
+
+        if type_featured_request == 'accepted':
+            queryset = queryset.filter(featured__isnull=False)
+        else:
+            queryset = queryset.filter(featured__isnull=True)
+
+        queryset = queryset.all()
+
+        return [q for q in queryset if isinstance(q.content_object, Topic) or not q.content_object.is_obsolete]
+
+    @method_decorator(login_required)
```

Comment:
2 remarques : 

- Je préfère largement l'utilisation de `@method_decorator(DECORATOR, name='dispatch')` lorsqu'on ne met rien dans la méthode en question, c'est plus propre à écrire et lire (cf https://docs.djangoproject.com/fr/1.11/topics/class-based-views/intro/#decorating-the-class)
- Tu peux utiliser le mixin `django.contrib.auth.mixins.LoginRequiredMixin` https://docs.djangoproject.com/en/1.11/topics/auth/default/#the-loginrequired-mixin et celui pour les permissions https://docs.djangoproject.com/en/1.11/topics/auth/default/#the-permissionrequiredmixin-mixin (ça permet d'avoir de jolies CBV)

Ce commentaire s'applique également à la suite du fichier